soccer/src/soccer/strategy/agent/position/position.cpp

```diff
@@ -286,4 +286,54 @@ communication::Acknowledge Position::acknowledge_ball_in_transit(
     return acknowledge_response;
 }
 
+// Checks whether ball is out of range for stealing/receiving
+bool Position::ball_in_red() const {
+    auto& ball_pos = last_world_state_->ball.position;
+    return (field_dimensions_.our_defense_area().contains_point(ball_pos) ||
+            field_dimensions_.their_defense_area().contains_point(ball_pos) ||
+            !field_dimensions_.field_rect().contains_point(ball_pos));
+}
+
+bool Position::can_steal_ball() const {
```

Comment:
i see now, they were elevated to superclass? then we should definitely delete from subclass.